app/models/round.rb

```diff
@@ -133,6 +133,25 @@ def advancement_condition_to_s(short: false)
     advancement_condition ? advancement_condition.to_s(self, short: short) : ""
   end
 
+  def competitors_live_results_entered
+    live_results.count
+  end
+
+  def score_taking_done?
+    is_open && competitors_live_results_entered == total_registrations
+  end
+
+  def round_can_be_opened?
+    return false if is_open
+    return true if number == 1
+    previous_round = Round.joins(:competition_event).find_by(competition_event: { competition_id: competition_event.competition_id, event_id: event.id }, number: number - 1)
+    previous_round.score_taking_done?
+  end
```

Comment:
Isn't this just `number == 1 || (!is_open && ...)`?

Reply:
no. that would always return true for rounds with `number == 1`. But these are not openable if they are already opened. it actually used to be this
```
def round_can_be_opened?
    return false if is_open
    return !is_open if number == 1
    previous_round = Round.joins(:competition_event).find_by(competition_event: { competition_id: competition_event.competition_id, event_id: event.id }, number: number - 1)
    previous_round.score_taking_done?
  end
```
but that `!is_open `is redundant after the first check